packages/core/src/components/service-message/ServiceMessageProvider.tsx

```diff
@@ -0,0 +1,23 @@
+import React,{ PropsWithChildren, useContext } from "react";
+import { useFetchServiceMessage } from "./hooks/useFetchServiceMessage";
+
+const ServiceMessageContext = React.createContext<ReturnType<typeof useFetchServiceMessage>>({
+    serviceMessage: undefined,
+    isDismissed: false,
+    setIsDismissed: () => undefined,
+    expansionEnabled: false,
+    setExpansionEnabled: () => undefined,
+    isExpanded: false,
+    setIsExpanded: () => undefined,
+    isError: false
+});
+
+export const ServiceMessageProvider = ({children}:PropsWithChildren) => {
```

Comment:
Any specific reason to use a provider here? If we only render the banner in the custom navigation header, do we need to have this added complexity? Seems most of it can be moved directly into the banner component as well.

Reply:
We need some kind of global state management here. Otherwise the state will be reset every time you go to a new screen

Reply:
Wouldn't this only happen if the component remounts on every navigation? Is this how it is working today?

Reply:
The component will remount. We add a service message banner to all screens by default, but they are separate entities

Reply:
Ahh. Thought the banner was stayed mounted during screen navigation